cli/repl.rs

```diff
@@ -222,11 +222,15 @@ impl LineHighlighter {
       (?P<comment>(?:/\*[\s\S]*?\*/|//[^\n]*)) |
       (?P<string>(?:"([^"\\]|\\.)*"|'([^'\\]|\\.)*'|`([^`\\]|\\.)*`)) |
       (?P<regexp>/(?:(?:\\/|[^\n/]))*?/[gimsuy]*) |
-      (?P<number>\d+(?:\.\d+)*(?:e[+-]?\d+)*n?) |
+      (?P<number>\b\d+(?:\.\d+)?(?:e[+-]?\d+)*n?\b) |
+      (?P<infinity>\b(?:Infinity|NaN)\b) |
+      (?P<hexnumber>\b0x[a-fA-F0-9]+\b) |
+      (?P<octalnumber>\b0o[0-7]+\b) |
+      (?P<binarynumber>\b0b[01]+\b) |
       (?P<boolean>\b(?:true|false)\b) |
       (?P<null>\b(?:null)\b) |
       (?P<undefined>\b(?:undefined)\b) |
-      (?P<keyword>\b(?:await|async|var|let|for|if|else|in|of|class|const|function|yield|return|with|case|break|switch|import|export|new|while|do|throw|catch)\b) |
+      (?P<keyword>\b(?:await|async|var|let|for|if|else|in|of|class|const|function|yield|return|with|case|break|switch|import|export|new|while|do|throw|catch|this)\b) |
```

Comment:
This was left out on purpose.

Editors tend to disagree here so up for debate, but I personally at-least tend to not highlight intrinsic values and identifiers.

Reply:
> Editors tend to disagree here

well, nano and vim treat this as a keyword (meaning giving it the same color as keywords)
sublime text and vscode gives it a special color
and all text editors (as i know) where cli or not highlight "this" in some way

nano:
![nano](https://user-images.githubusercontent.com/48116123/97914117-0a2d4a80-1d58-11eb-8c0c-e29fa1443e33.png)
sublime:
![sumblime_text](https://user-images.githubusercontent.com/48116123/97914120-0b5e7780-1d58-11eb-89d9-fc83b11b765c.png)
vim:
![vim](https://user-images.githubusercontent.com/48116123/97914123-0bf70e00-1d58-11eb-945b-ca493848d883.png)
vscode:
![vscode](https://user-images.githubusercontent.com/48116123/97914124-0bf70e00-1d58-11eb-9051-5d665dcb0e42.png)

>  but I personally at-least tend to not highlight intrinsic values and identifiers. If we do it too much the entire output lights up a bright yellow

highlighting "this" only is not too much

Reply:
I agree with @AliBasicCoder on this one - I've never seen `this` not being highlighted.